app/schema/widgets/currency_widget.py

```diff
@@ -0,0 +1,30 @@
+from app.schema.widget import Widget
+from flask import render_template
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class CurrencyWidget(Widget):
+    def __init__(self, name):
+        super().__init__(name)
+
+    def render(self, schema, state):
+        widget_params = {
+            'answer': {
```

Comment:
Am waiting for Hamish to commit his changes to the DateWidget, and for me to finish the checkbox and radio group widgets.  Once those are in it would make sense to try refactor the interface between python and html.